### PR TITLE
ocaml 4.10.2

### PR DESCRIPTION
**Pull request by kit-ty-kate**
Brings native macOS/arm64 support

### CODE REVIEW

**Comment by kit-ty-kate** [comment]
mmh, your SPDX format detector doesn't seem to support [license exceptions](https://spdx.org/licenses/exceptions-index.html). Is that expected?

File: Formula/ocaml.rb [hunk 1]
```diff
@@ -13,9 +13,9 @@
 class Ocaml < Formula
   desc "General purpose programming language in the ML family"
   homepage "https://ocaml.org/"
-  url "https://caml.inria.fr/pub/distrib/ocaml-4.10/ocaml-4.10.0.tar.xz"
-  sha256 "30734db17b609fdd1609c39a05912325c299023968a2c783e5955dd5163dfeb7"
-  license "LGPL-2.1"
+  url "https://caml.inria.fr/pub/distrib/ocaml-4.10/ocaml-4.10.2.tar.xz"
+  sha256 "96871461078282d5db022077d89bde25c85fb5e376612b44f8c37d4e84f000e3"
+  license "LGPL-2.1-only WITH OCaml-LGPL-linking-exception"
```
Comment (SMillerDev) [inline Formula/ocaml.rb:18]:
```suggestion
  license "LGPL-2.1-only" => { with: "OCaml-LGPL-linking-exception" }
```

Comment (kit-ty-kate) [reply Formula/ocaml.rb]: oh I see. Thanks!

**Comment by kaepora** [comment]
Thanks!

**Comment by kit-ty-kate** [comment]
I'm unsure what are the failures about. What did I forget to do?

**Comment by kit-ty-kate** [comment]
I believe https://github.com/Homebrew/formula-patches/pull/318 should be removed if this gets merged. cc @claui

**Comment by SMillerDev** [comment]
Please don't use merge commits, it trips up our CI.

**Comment by carlocab** [comment]
One commit per formula file change, please.

**Comment by chenrui333** [comment]
close in favor of #67721, thanks @kit-ty-kate!

**Pull request closed without merging**